From 4e9fcf418c5432d1af08bf833d54fb8a6be52a92 Mon Sep 17 00:00:00 2001
From: Gustavo Duarte <gduarte@redhat.com>
Date: Tue, 4 Nov 2014 22:07:14 -0500
Subject: [powerpc] numa: check error return from proc_create

Message-id: <1415138837-28185-6-git-send-email-gduarte@redhat.com>
Patchwork-id: 99394
O-Subject: [RHEL7.1 PATCH BZ 1150097 5/8] powerpc/numa: check error return from proc_create
Bugzilla: 1150097
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Don Zickus <dzickus@redhat.com>
RH-Acked-by: Steve Best <sbest@redhat.com>

BZ 1150097
Upstream Status: 2d15b9b479512f05680541acffd9acbbc831a47c

commit 2d15b9b479512f05680541acffd9acbbc831a47c
Author: Nishanth Aravamudan <nacc@linux.vnet.ibm.com>
Date:   Thu Oct 9 16:41:28 2014 -0700

    powerpc/numa: check error return from proc_create

    proc_create can fail, we should check the return value and pass up the
    failure.

    Suggested-by: Michael Ellerman <mpe@ellerman.id.au>
    Signed-off-by: Nishanth Aravamudan <nacc@linux.vnet.ibm.com>
    Signed-off-by: Michael Ellerman <mpe@ellerman.id.au>

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/powerpc/mm/numa.c b/arch/powerpc/mm/numa.c
index 1487514..17c65b6 100644
--- a/arch/powerpc/mm/numa.c
+++ b/arch/powerpc/mm/numa.c
@@ -1810,7 +1810,8 @@ static const struct file_operations topology_ops = {
 static int topology_update_init(void)
 {
  start_topology_update();
- proc_create("powerpc/topology_updates", 0644, NULL, &topology_ops);
+ if (!proc_create("powerpc/topology_updates", 0644, NULL, &topology_ops))
+  return -ENOMEM;
 
  return 0;
 }
-- 
1.7.1